src/coprocessor/dag/expr/constant.rs

```diff
@@ -3,17 +3,72 @@
 use std::borrow::Cow;
 
 use super::{Constant, Result};
+use crate::coprocessor::codec::convert::{
+    convert_bytes_to_int, convert_datetime_to_int, convert_decimal_to_int, convert_duration_to_int,
+    convert_float_to_int, convert_json_to_int,
+};
 use crate::coprocessor::codec::mysql::{Decimal, Duration, Json, Time};
-use crate::coprocessor::codec::Datum;
+use crate::coprocessor::codec::{Datum, Error};
+use crate::coprocessor::dag::expr::EvalContext;
+use cop_datatype::FieldTypeTp;
 
 impl Datum {
     #[inline]
-    pub fn as_int(&self) -> Result<Option<i64>> {
+    pub fn as_bool(&self, ctx: &mut EvalContext) -> Result<Option<bool>> {
```

Comment:
I think we should implement `AsMySQLBool` trait for `Datum` instead of adding an extra method to it.

Reply:
The AsMySQLBool funct is `fn as_mysql_bool(&self, context: &mut EvalContext) -> Result<bool>;`, its `Result` is `tikv::coprocessor::error`. The Datum's `as_*`'s `Result<Option<x>>`'s Err is `tikv::coprocessor::codec::error`

Reply:
The `tikv::coprocessor::codec::error` will convert to `tikv::coprocessor::error` eventually.